third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/ScanOpToLLVM.cpp

```diff
@@ -439,6 +440,25 @@ unpackInputs(Location loc, triton::ScanOp op, triton::ScanOpAdaptor adaptor,
   return srcValues;
 }
 
+SmallVector<SmallVector<Value>> 
+flipSrcValues(Location loc, triton::ScanOp op, 
+                   ConversionPatternRewriter &rewriter,
+                   SmallVector<SmallVector<Value>> srcValues,
+                   int iWarpSize){
+    SmallVector<SmallVector<Value>> values(srcValues.size());
+    for (int i = 0; i < srcValues.size(); ++i) {
+        int revIndex = srcValues.size() - i - 1;
+        for (unsigned j = 0; j < op.getNumOperands(); ++j) {
+            for (unsigned k = iWarpSize / 2; k >= 1; k = k / 2){
+                srcValues[revIndex][j] = shflSync(loc, rewriter, srcValues[revIndex][j], k);
```

Comment:
Separately I reverse the warp indices when the axis is distributed, which I think handles this case.

Reply:
Instead of using `shflSync`. I think reverse srcIndex and employ shflDownSync might be even more efficient, but the current solution also seems fine to me because it modifies less code.

Reply:
I tried pretty hard to get `shflDownSync` to work but I ended up having to add a lot of the complex cross warp code (if reverse switch first/last etc). This seems more maintainable.

Reply:
np. Could you please drop a comment?